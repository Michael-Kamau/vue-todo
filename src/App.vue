<template>
  <div :class="{ dark: darkThemeEnabled }">
    <Home/>
  </div>
</template>
<script>

import Home from './components/Home.vue'
import {mapGetters} from "vuex";

export default {
  components: {Home},
  computed: {
    ...mapGetters({
      'darkThemeEnabled': 'darkThemeEnabled',
    })
  },

}
</script>

<style scoped>

</style>
