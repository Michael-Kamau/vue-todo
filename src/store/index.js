import {createApp} from 'vue'
import {createStore} from 'vuex'
import todos from "./modules/todos.js";

// Create a new store instance.
const store = createStore({
    state() {
        return {
        }
    },
    mutations: {},

    modules:{
        todos
    }
})

export default store
