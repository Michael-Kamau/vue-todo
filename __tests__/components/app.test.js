import {assert, describe, it, expect, beforeEach} from 'vitest'
import {render, screen} from "@testing-library/vue";
import { mount } from '@vue/test-utils'

import App from "../../src/App.vue";
import store from "../../src/store/index.js";


describe('suite', () => {
    let wrapper

    beforeEach(() => {
        wrapper = render(App, {
            global: {
                plugins: [store]
            }
        });
    })

    it("snap shot matches", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('A todo item can be added to the list', async () => {
        const wrapper = mount(App,{
            global: {
                plugins: [store]
            }
        })
        const todoInput = wrapper.find('#todoInput')

        todoInput.setValue('New todo item');
        await todoInput.trigger('keyup.enter');

        const itemsText = wrapper.find('#itemsLeft')

        expect(todoInput.exists()).toBe(true)
        expect(itemsText.text()).toBe('1 items left')
        expect(todoInput.text()).toBe('')

    })

    it('A todo item can be marked as complete', async () => {
        const wrapper = mount(App,{
            global: {
                plugins: [store]
            }
        })
        const todoItem = wrapper.find('.item')
        const itemsText = wrapper.find('#itemsLeft')
        const completeButton = wrapper.find('#completeButton')


        expect(todoItem.exists()).toBe(true)
        expect(completeButton.exists()).toBe(true)
        expect(todoItem.text()).toBe('New todo item')

        await completeButton.trigger('click')


        expect(itemsText.text()).toBe('0 items left')

    })


    it('A user can clear all completed items', async () => {
        const wrapper = mount(App,{
            global: {
                plugins: [store]
            }
        })
        const clearComplete = wrapper.find('#clearComplete')


        expect(clearComplete.exists()).toBe(true)
        await clearComplete.trigger('click')

        const todoItem = wrapper.find('.item')
        expect(todoItem.exists()).toBe(false)


    })



})
