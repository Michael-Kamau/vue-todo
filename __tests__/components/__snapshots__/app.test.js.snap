// Vitest Snapshot v1

exports[`suite > snap shot matches 1`] = `
{
  "baseElement": <body>
    <div>
      <div
        class="font-rubik"
      >
        <div
          class="flex flex-col bg-gray-100 dark:bg-gray-900 min-h-screen"
        >
          <div
            class="h-52 w-full bg-cover bg-no-repeat bg-mobile-light dark:bg-mobile-dark md:bg-desktop-light dark:md:bg-desktop-dark absolute"
          />
          <div
            class="w-5/6 sm:w-2/3 md:w-2/4 lg:w-1/3 mx-auto py-4 z-10"
          >
            <div
              class="flex justify-between items-center"
            >
              <h1
                class="text-white text-3xl font-bold my-8"
              >
                TODO
              </h1>
              <div
                class="w-6 h-7 bg-no-repeat bg-moon-icon dark:bg-sun-icon cursor-pointer"
              />
            </div>
            <div
              class="relative text-gray-600 focus-within:text-gray-400"
            >
              <span
                class="absolute inset-y-0 left-0 flex items-center pl-4"
              >
                <div
                  class="flex items-center justify-center h-5 w-5 rounded-full bg-gray-400 hover:bg-gradient-to-r hover:from-sky-400 hover:to-violet-500 cursor-pointer"
                >
                  <div
                    class="bg-white dark:bg-gray-800 h-[1.15rem] w-[1.15rem] rounded-full m-auto"
                  />
                </div>
              </span>
              <input
                class="form-control block w-full mx-auto my-1 pr-3 pl-10 py-4 text-base font-normal text-gray-600 dark:text-gray-300 dark:bg-gray-800 rounded transition ease-in-out m-0 dark:focus:text-gray-300 dark:focus:bg-gray-800 focus:outline-none"
                id="todoInput"
                placeholder="Create a new todo..."
                type="text"
              />
            </div>
            <section
              class="w-full mx-auto my-5"
              data-v-de4ea765=""
            >
              <div
                class="divide-y divide-gray-200 dark:divide-gray-600"
                data-v-de4ea765=""
              >
                <transition-group-stub
                  appear="false"
                  class="divide-y divide-gray-200 dark:divide-gray-600"
                  css="true"
                  data-v-de4ea765=""
                  name="todos"
                  persisted="false"
                  tag="div"
                >
                  
                  
                </transition-group-stub>
                <div
                  class="flex justify-between items-center py-4 px-2 sm:px-4 dark:text-gray-300 bg-white dark:bg-gray-800 text-gray-400 dark:text-gray-300 rounded-b text-xs font-semibold"
                  data-v-de4ea765=""
                >
                  <p
                    class="font-normal"
                    data-v-de4ea765=""
                    id="itemsLeft"
                  >
                    0 items left
                  </p>
                  <div
                    class="flex space-x-1 sm:space-x-2 sm:mx-1"
                    data-v-de4ea765=""
                  >
                    <p
                      class="cursor-pointer hover:text-gray-500 dark:hover:text-gray-100 hover:scale-105 text-blue-400"
                      data-v-de4ea765=""
                    >
                       All
                    </p>
                    <p
                      class="cursor-pointer hover:text-gray-500 dark:hover:text-gray-100 hover:scale-105"
                      data-v-de4ea765=""
                    >
                      Active
                    </p>
                    <p
                      class="cursor-pointer hover:text-gray-500 dark:hover:text-gray-100 hover:scale-105"
                      data-v-de4ea765=""
                    >
                      Completed
                    </p>
                  </div>
                  <p
                    class="cursor-pointer hover:text-gray-500 dark:hover:text-gray-100 font-normal"
                    data-v-de4ea765=""
                  >
                     Clear Completed
                  </p>
                </div>
              </div>
            </section>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="font-rubik"
    >
      <div
        class="flex flex-col bg-gray-100 dark:bg-gray-900 min-h-screen"
      >
        <div
          class="h-52 w-full bg-cover bg-no-repeat bg-mobile-light dark:bg-mobile-dark md:bg-desktop-light dark:md:bg-desktop-dark absolute"
        />
        <div
          class="w-5/6 sm:w-2/3 md:w-2/4 lg:w-1/3 mx-auto py-4 z-10"
        >
          <div
            class="flex justify-between items-center"
          >
            <h1
              class="text-white text-3xl font-bold my-8"
            >
              TODO
            </h1>
            <div
              class="w-6 h-7 bg-no-repeat bg-moon-icon dark:bg-sun-icon cursor-pointer"
            />
          </div>
          <div
            class="relative text-gray-600 focus-within:text-gray-400"
          >
            <span
              class="absolute inset-y-0 left-0 flex items-center pl-4"
            >
              <div
                class="flex items-center justify-center h-5 w-5 rounded-full bg-gray-400 hover:bg-gradient-to-r hover:from-sky-400 hover:to-violet-500 cursor-pointer"
              >
                <div
                  class="bg-white dark:bg-gray-800 h-[1.15rem] w-[1.15rem] rounded-full m-auto"
                />
              </div>
            </span>
            <input
              class="form-control block w-full mx-auto my-1 pr-3 pl-10 py-4 text-base font-normal text-gray-600 dark:text-gray-300 dark:bg-gray-800 rounded transition ease-in-out m-0 dark:focus:text-gray-300 dark:focus:bg-gray-800 focus:outline-none"
              id="todoInput"
              placeholder="Create a new todo..."
              type="text"
            />
          </div>
          <section
            class="w-full mx-auto my-5"
            data-v-de4ea765=""
          >
            <div
              class="divide-y divide-gray-200 dark:divide-gray-600"
              data-v-de4ea765=""
            >
              <transition-group-stub
                appear="false"
                class="divide-y divide-gray-200 dark:divide-gray-600"
                css="true"
                data-v-de4ea765=""
                name="todos"
                persisted="false"
                tag="div"
              >
                
                
              </transition-group-stub>
              <div
                class="flex justify-between items-center py-4 px-2 sm:px-4 dark:text-gray-300 bg-white dark:bg-gray-800 text-gray-400 dark:text-gray-300 rounded-b text-xs font-semibold"
                data-v-de4ea765=""
              >
                <p
                  class="font-normal"
                  data-v-de4ea765=""
                  id="itemsLeft"
                >
                  0 items left
                </p>
                <div
                  class="flex space-x-1 sm:space-x-2 sm:mx-1"
                  data-v-de4ea765=""
                >
                  <p
                    class="cursor-pointer hover:text-gray-500 dark:hover:text-gray-100 hover:scale-105 text-blue-400"
                    data-v-de4ea765=""
                  >
                     All
                  </p>
                  <p
                    class="cursor-pointer hover:text-gray-500 dark:hover:text-gray-100 hover:scale-105"
                    data-v-de4ea765=""
                  >
                    Active
                  </p>
                  <p
                    class="cursor-pointer hover:text-gray-500 dark:hover:text-gray-100 hover:scale-105"
                    data-v-de4ea765=""
                  >
                    Completed
                  </p>
                </div>
                <p
                  class="cursor-pointer hover:text-gray-500 dark:hover:text-gray-100 font-normal"
                  data-v-de4ea765=""
                >
                   Clear Completed
                </p>
              </div>
            </div>
          </section>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
